File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        int i=0, j=0,k=0;

        while(k<goal.length ){
            
            if(cards1[i<cards1.length? i:cards1.length-1].equals(goal[k])){
                k++;
                i++;
            }else if(cards2[j<cards2.length? j:cards2.length-1].equals(goal[k])){
                k++;
                j++;
            }else{
                return "No";
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var i = 0
        var j = 0
        var k = 0

        while (k < goal.size) {
            if (i < cards1.size && cards1[i] == goal[k]) {
                i++
            } else if (j < cards2.size && cards2[j] == goal[k]) {
                j++
            } else {
                return "No"
            }
            k++
        }
        return "Yes"
    }
}
